allennlp/models/constituency_parser.py

```diff
@@ -326,11 +305,11 @@ def assemble_subtree(start: int, end: int):
             # This node is a leaf.
             if end - start == 1:
                 word = sentence[start]
-                tree = {"start": start, "end": end, "word": word, "is_leaf": True}
                 if label is not None:
-                    tree["label"] = label
-                if pos_tags is not None:
-                    tree["pos_tag"] = pos_tags[start]
+                    tree = Tree(label, [word])
+                    if pos_tags is not None:
+                        # Tree has pre-terminal POS tags.
+                        tree = Tree(label, [Tree(pos_tags[start], [word])])
                 return [tree]
```

Comment:
Looks like this will crash if `label` is `None`.  Is it possible to get into that state?  If it shouldn't be possible, you could `assert label is not None, "message"`, and remove the `if`.

Reply:
Thanks, this was actually wrong. Now, if POS tags are explicitly provided and the node has a label, we add the POS tags as pre-terminal nodes. If there is no POS tag but there are labels, we just use the label. If there is no label and no POS tags, we use a dummy XX POS as NLTK doesn't allow unlabelled nodes.